crawl-ref/source/beam.cc

```diff
@@ -3158,6 +3158,21 @@ void bolt::affect_player_enchantment(bool resistible)
                 mprf("You%s", you.resist_margin_phrase(margin).c_str());
             }
         }
+        
+        // punish the monster if we're a willful demon
+        if (you.get_mutation_level(MUT_DEMONIC_WILL))
+        {
+            monster* mon = monster_by_mid(source_id);
+            if (mon && mon->alive())
+            {
+                int damage_roll = 3 + random2(11);
```

Comment:
Maybe this should scale by beam power? Damaging more powerful hexers more heavily would make it more relevant. As it is, it feels like a very pale shadow of Ru aura and I feel like players won't notice it unless it deals the killing blow.

Reply:
Implemented. Kept the power level fairly low since this is the rank 1 and it also provides will+ though.